htdocs/core/class/commonobject.class.php

```diff
@@ -320,11 +341,11 @@ abstract class CommonObject
 	public $newref;
 
 	/**
-	 * @var int 		The object's status. Use status instead.
+	 * @var int	Deprecated property. Use $status instead.
 	 * @deprecated
-	 * @see setStatut()
+	 * @see $status
 	 */
-	public $statut;
+	private $statut;
```

Comment:
A lot of field are deprecated since a short time, so we can't remove their usability too quickly. If property is set to private, all external modules and custom dev will be broken without having enough time for a compatibility period.

Reply:
> A lot of field are deprecated since a short time, so we can't remove their usability too quickly. If property is set to private, all external modules and custom dev will be broken without having enough time for a compatibility period.

The fields still exist as virtual properties - the private fields are only there to help the IDE - they were added for that purpose.

The DolDeprecationHandler will serve the '$status' property when '$statut' is accessed and update the '$status' property when '$statut' is updated.  It avoid any need to have double assignments in the code where we see "$this->status = $something; $this->statut = $something;" .
IMHO these double assignments are easily forgotten but with this Trait that would no longer be an issue.

At the same time as the deprecated properties are virtual any access is captured and reported as a deprecated access.

The PHPUNIT test demonstrates this: - one example where the oldPropertyName is assigned and the value is read back through the new property name:

https://github.com/Dolibarr/dolibarr/blob/f2001cb05449161d3b7cf61f0004ef6d1a602e06/test/phpunit/DolDeprecationHandlerTest.php#L241-L248

Reply:
I don't understand. 
If the code of an external module do:
$theobject->statut = 1;
On any object that etends commobject.class.php (so any business object), if property statut is set as private, this will generate a php warning/error. Did i miss something.

Reply:
The deprecation warnings are only generated when the deprecation detection is on.
So that is not on in a production environment and should not make things fail.

The "trigger_error" calls are gated with 
https://github.com/Dolibarr/dolibarr/blob/492efeeaacdd3de71c21cd0632dd52a4c97d0d2c/htdocs/core/class/doldeprecationhandler.class.php#L158-L161

and the value return by an optional function in the class the Trait is attached to:
`isDynamicPropertiesEnabled()`.

When the latter is set, access to dynamic properties trigger a notice.

In a test setup the deprecation warnings are active and certain accesses will result in a failing test.

This can be observed in the current test results.

There can be several steps: 
1. Integrate DolDeprecatedHandler with "isDynamicPropertiesEnabled()" returning "true" by default (the tests should pass);
2. Create a new PR with isDynamicPropertiesEnabled() returning false by default (the same tests will fail);
3. In the futur attach DolDeprecatedHandler progressively to other classes/extend the deprecation mapping in subclasses and remove the deprecated definitions and explicit assignments to thos definitions.